### DISCUSSION TITLE
Spring for Graphql - custom scalar for Set

### DISCUSSION THREAD

**Discussion by leandrocotsa** [graphql-java/graphql-java#2890] [Q&A]
I'm using spring for graphql and I'm having trouble with the Set data structure as an input for my graphql queries.
Basically, if I send a [...] for Sets I get an error of ArrayList being not assignable to Set.

With other graphql version I used previously, Jackson would automatically convert it to a Set. But according to spring for graphql docs:  
_Unlike web frameworks, GraphQL does not use Jackson annotations to drive JSON serialization/deserialization. Custom data types and their serialization must be described as Scalars._

I tried to implement my own Set scalar/coercing following a similiar structure to this scalar that handles maps, but I can't make it work for Sets:

https://github.com/graphql-java/graphql-java-extended-scalars/blob/master/src/main/java/graphql/scalars/object/ObjectScalar.java

Can anyone help me?

**Comment by bbakerman** [comment]
Can you give more details please - a stack trace, a type system snippet, a query snippet.

There is too little information above for us to meaningfully help.
